crates/rolldown_binding/src/options/binding_output_options/mod.rs

```diff
@@ -1,12 +1,11 @@
 use super::super::types::binding_rendered_chunk::RenderedChunk;
 use super::plugin::BindingPluginOrParallelJsPluginPlaceholder;
-use crate::types::js_callback::MaybeAsyncJsCallback;
 use derivative::Derivative;
 use napi::threadsafe_function::ThreadsafeFunction;
 use napi_derive::napi;
 use serde::Deserialize;
 
-pub type AddonOutputOption = MaybeAsyncJsCallback<RenderedChunk, Option<String>>;
+pub type AddonOutputOption = ThreadsafeFunction<RenderedChunk, Option<String>, false>;
```

Comment:
```suggestion
pub type AddonOutputOption = JsCallback<RenderedChunk, Option<String>, false>;
```

Reply:
Use `JsCallback` here. Could you also add the reason that why change it from async to sync. I know there's problem of footer and banner. Would you mind describing how this fix it?

Reply:
Here need to be sync at first. Other reason here change it because the result of `MaybeAsyncJsCallback` need to be a `Promise<T>`, if the result is `String` will error at here, The `Either<Either<Promise<Ret>, Ret>` look like is not correct. 
The `JsCallback` need to be implement `await_call` method, it is unnecessary. Actually, I think here `JsCallback` using `ThreadsafeFunction<Args, Ret, false>` is enough, the `Either<Ret, UnknownReturnValue>` is unnecessary.

Reply:
> The JsCallback need to be implement await_call method

I don't understand what you are saying here.`JsCallback` is just a type alias. It has nothing to do with `implement await_call method`.

The reason we use type alias is to make devs understand easily these functions are connected with JavaScript while `threadafe_function` is a little bit difficult to understand in the first look.

To keep the code style and be friendly to contributors, I suggest you to use `JsCallback` here. It has a [full page comment](https://github.com/rolldown/rolldown/blob/a43d3b5f5c3c405404934080124e7ac929e5150a/crates/rolldown_binding/src/types/js_callback.rs#L10-L68) to describe how to understand it for contributors.

> if the result is String will error at here, The Either<Either<Promise<Ret>, Ret> look like is not correct.

If the `ThreadsafeFunction<(),  Either<Either<Promise<String>, String>>` couldn't handle `returning 'hello'` from js side, it might be an unexpected behavior of NAPI-RS.

It's better that you point out this in the PR description and we notify NAPI-RS. Then we fix it by creating an issue that this fix is a workaround, we should revert the change if NAPI-RS fix the behavior. @Brooooooklyn cc

> Actually, I think here JsCallback using ThreadsafeFunction<Args, Ret, false> is enough, the Either<Ret, UnknownReturnValue> is unnecessary.

1. All code in this repo is so far written with purpose. If you think something is unnecessary, please provide a reason.
2. Old NAPI-RS doesn't throw friendly errors like it does now. So this `Either<Ret, UnknownReturnValue>` could help us to quickly find out what's wrong.
3. Even though NAPI-RS now emit the error like ` Failed to convert JavaScript value `Number 1 ` into rust type `String`, still the `Either<Ret, UnknownReturnValue>` here could help us quickly locate the convertion error is happed while returning something from js, not args or struct fields.

Reply:
Keep the `Either<..., Unkowwn>` . The best error message we can get for debugging unexpected passed js value is

```
Caused by: Error: Rolldown internal error: InvalidArg, Unknown return value. Cannot convert to `Option<String>` in `ThreadsafeFunction<RenderedChunk, Either<Either<Promise<Option<String>>, Option<String>>, UnknownReturnValue>, false>`.
```

This is much friendly that the raw error message that NAPI-RS prodvided.

Reply:
> This is much friendly that the raw error message that NAPI-RS prodvided.
OK. I get it.

> Here need to be sync at first.
I find i had a mistake for it after changed typing. Oirginal is correct.

> The JsCallback need to be implement await_call method
I will descript it at new pr.

Reply:
> Caused by: Error: Rolldown internal error: InvalidArg, Unknown return value. Cannot convert to `core::option::Option<alloc::string::String>`.

. The error is also difficult to debugger. I meet it at today, i can't find anywhere crashed. I think we can add fields  `name` to make it easy.

Reply:
Yeah. It's better than the old NAPI-RS, but indeed still confusing. So that's what https://github.com/rolldown/rolldown/pull/970 try to solve.